app/util/event_log.py

```diff
@@ -0,0 +1,265 @@
+import collections
+import json
+from logbook import RotatingFileHandler, StreamHandler
+from logbook.more import TaggingHandler, TaggingLogger
+import os
+import sys
+import time
+
+from app.util import fs, log
+from app.util.conf.configuration import Configuration
+from app.util.counter import Counter
+
+
+LOG_CACHE_EXPIRE_TIME_IN_HOURS = 5
+LOG_CACHE_THRESHOLD = 10000
```

Comment:
What do you think of cranking this up to like 50k or 100k? I just loaded our timeline dashboard and it is not even super tightly packed but still loaded 12k events. (Each of our PHPUnit jobs has ~1k subjobs, with each subjob having 2 events each, so it would only take 5 phpunit jobs within half an hour to surpass a cache of 10k.)